src/odemis/driver/xt_client.py

```diff
@@ -67,6 +69,108 @@
+
+    @property
+    def adapter(self):
+        return self._adapter
+
+    @adapter.setter
+    def adapter(self, value):
+        self._adapter = value
+
+    @property
+    def bitness(self):
+        return self._bitness
+
+    @bitness.setter
+    def bitness(self, value):
+        self._bitness = value
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    @property
+    def path(self):
+        return self._path
+
+    @path.setter
+    def path(self, value):
+        self._path = value
+
+    @property
+    def version(self):
+        return self._version
+
+    @version.setter
+    def version(self, value):
+        self._version = value
+
+
+def check_latest_package(
+    directory: str, current_version: str, adapter: str, bitness: str, is_zip: bool
+) -> Package:
+    """
+    Check if a latest xtadapter package is available.
+
+    :param directory: The directory to check if a latest xtadapter package is available.
+    :param current_version: The currently installed xtadapter version.
+    :param adapter: The adapter type e.g. xtadapter, fastem-xtadapter to be used for filtering.
+    :param bitness: The executable bitness i.e. 32bit or 64bit to be used for filtering.
+    :param is_zip: A flag stating if the package is a zip file.
+    :return: A class containing relevant information about the latest xtadapter package, if not found return Package().
+
+    """
+    # Dict where key is the version of xtadapter package and value is information about the package
+    version_package = {}
+    # Package is named as f"delmic-{adapter}-{bitness}bit-v{version}"
+    # delmic-xtadapter-32bit-v1.11.2-dev
+    # delmic-xtadapter-32bit-v1.11.2.zip
+    # delmic-fastem-xtadapter-64bit-v1.11.2-dev
+    regex = r"delmic-([a-zA-Z-]+)-(\d+)bit-v([\d.]+)"
+    if is_zip:
+        regex += r".*.zip"
+    if os.path.isdir(directory):
+        for entity in os.listdir(directory):
+            result = re.search(regex, entity)
+            if result and (is_zip == entity.endswith(".zip")):
+                package = Package()
+                package.adapter = result.group(1)
+                package.bitness = result.group(2) + "bit"
+                package.version = result.group(3)
+                package.name = entity
+                package.path = os.path.join(directory, entity)
+                # Filter using adapter type and bitness
+                if bitness == package.bitness and adapter == package.adapter:
+                    version_package[package.version] = package
+        versions = list(version_package)
+        if len(versions):
+            # Sort the available versions in reverse order i.e. the latest version is the first element
+            versions.sort(key=lambda s: [int(u) for u in s.split(".")], reverse=True)
+            if current_version < versions[0]:
+                print("The latest version is {}.\n".format(versions[0]))
```

Comment:
why do you use a print statement here?

Reply:
So that the user first gets to read the print in the application window, next the message box pops-up asking yes/no question for the update.

Reply:
Don't expect the driver's standard output to be visible (typically it's hidden). Just use `logging` for any output.